requirements/updater.sh

```diff
@@ -1,62 +1,69 @@
 #!/usr/bin/env bash
+# -*- coding: utf-8; mode: sh; sh-indentation: 2; indent-tabs-mode: nil; sh-basic-offset: 2; -*-
+
 set -ue
 
 PYTHON=python3.11
 
 for FILE in requirements.in requirements_all.txt ; do
-	if [ ! -f ${FILE} ] ; then
-		touch ${FILE}
-	fi
+  if [ ! -f ${FILE} ] ; then
+    touch ${FILE}
+  fi
 done
 requirements_dir="$(readlink -f .)"
 requirements_txt="$(readlink -f ./requirements_all.txt)"
 pip_compile="pip-compile --no-header --quiet -r --allow-unsafe"
 
 _cleanup() {
-  cd /
   test "${KEEP_TMP:-0}" = 1 || rm -rf "${_tmp}"
 }
 
 generate_requirements() {
-  venv="`pwd`/venv"
-  echo $venv
+  venv="$(pwd)/venv"
+  echo "Using virtual environment: $venv"
   ${PYTHON} -m venv "${venv}"
   # shellcheck disable=SC1090
   source ${venv}/bin/activate
 
-  ${venv}/bin/python -m pip install -U 'pip' pip-tools
+  ${venv}/bin/pip install -U pip pip-tools
+
+  ${pip_compile} ${requirements_dir}/requirements*.in --output-file requirements_all.txt
 
-  ${pip_compile} $(ls ${requirements_dir}/requirements*.in | xargs) --output-file requirements_all.txt
+  for file in ${requirements_dir}/requirements*.in; do
+    app="$(basename $file .in)"
+    echo "Compiling deps for $app"
+    ${pip_compile} $file --output-file "$app.txt" -c requirements_all.txt
+  done
 }
 
 main() {
   base_dir=$(pwd)
 
-  _tmp=$(${PYTHON} -c "import tempfile; print(tempfile.mkdtemp(suffix='.ansible_base-requirements', dir='/tmp'))")
+  _tmp=$(mktemp -d /tmp/tmp.XXXXXX.ansible_base-requirements)
```

Comment:
We did the python thing here because of subtle differences between mktmp on OS X and Fedora (and potentally other distros). For example, this mktmp command on OS X gives the following literal directory:
```
/tmp/tmp.XXXXXX.ansible_base-requirements
```

Whereas the same command on Fedora will actually put a value in there:
```
/tmp/tmp.3AqBMX.ansible_base-requirements
```

Reply:
Yeah on BSD/MacOS, the template has to come at the end. So `mktemp -d /tmp/ansible_base-requirements.XXXXXX` should work.


```suggestion
  _tmp=$(mktemp -d /tmp/ansible_base-requirements.XXXXXX)
```